sdk/textanalytics/ai-text-analytics/src/generated/generatedClientContext.ts

```diff
@@ -23,26 +20,17 @@ export class GeneratedClientContext extends coreHttp.ServiceClient {
    * @param options The parameter options
    */
   constructor(endpoint: string, options?: GeneratedClientOptionalParams) {
-    if (endpoint === undefined) {
-      throw new Error("'endpoint' cannot be null");
-    }
-
-    // Initializing default values for options
-    if (!options) {
-      options = {};
-    }
-
-    if (!options.userAgent) {
-      const defaultUserAgent = coreHttp.getDefaultUserAgentValue();
-      options.userAgent = `${packageName}/${packageVersion} ${defaultUserAgent}`;
-    }
-
-    super(undefined, options);
+    const defaults: GeneratedClientOptionalParams = {
+      requestContentType: "application/json; charset=utf-8"
+    };
 
-    this.requestContentType = "application/json; charset=utf-8";
+    const optionsWithDefaults = {
+      ...defaults,
+      ...options,
+      baseUri: endpoint + "text/analytics/v3.1-preview.3"
```

Comment:
do we ever have to worry about endpoint being changed dynamically? The old version would replace each time. Also I worry if endpoint doesn't end with a trailing slash this would break

Reply:
Ok I changed it to be set as follows:
```ts
baseUri:
        options?.endpoint ||
        (endpoint.slice(-1) === "/" ? endpoint : endpoint + "/") + "text/analytics/v3.1-preview.3"
```